### ISSUE TITLE
Graphics and visuals

### ISSUE THREAD

**Issue by tobyplowy** [LordStephen77/DreamBlocks-Beyond-of-Time#1]
This is a place where I can ask graphical related questions

**Comment by LordStephen77** [comment]
Here are the pictures I use for the project now. Now you have the clearest graphic style I want. I hope you like it. I prefer to make something different, instead of the usual Minecraft style.

Current player:

<a href='https://postimg.org/image/59fcmhwfj/' target='_blank'><img src='https://s3.postimg.org/ptk6kzc6r/body.png' border='0' alt="body"/></a><br/><br/>
<a href='https://postimg.org/image/i23gmf81b/' target='_blank'><img src='https://s3.postimg.org/5ngom3gj7/eyes.png' border='0' alt="eyes"/></a><br/><br/>
<a href='https://postimg.org/image/4ml1apoxb/' target='_blank'><img src='https://s3.postimg.org/vkeycg9kj/hairs.png' border='0' alt="hairs"/></a><br/><br/>
<a href='https://postimg.org/image/lbmh6mlin/' target='_blank'><img src='https://s3.postimg.org/ihjbt6jcj/head.png' border='0' alt="head"/></a><br/><br/>
<a href='https://postimg.org/image/9b115we3z/' target='_blank'><img src='https://s3.postimg.org/nhgs14oz7/PNJAtlas.png' border='0' alt="PNJAtlas"/></a><br/><br/>

Blocks textures:

<a href='https://postimg.org/image/48otvenz3/' target='_blank'><img src='https://s11.postimg.org/utrcqyqcj/materials.png' border='0' alt='materials'/>

<a href='https://postimg.org/image/dhoini481/' target='_blank'><img src='https://s24.postimg.org/i3kmvupr9/items.png' border='0' alt="items"/></a><br/><br/>
<a href='https://postimg.org/image/u6pyjf0td/' target='_blank'><img src='https://s24.postimg.org/mqqoxmd45/tools.png' border='0' alt="tools"/></a><br/><br/>

Backgrouds:

<a href='https://postimg.org/image/cgyw5y2sn/' target='_blank'><img src='https://s7.postimg.org/vytjlvzqj/back.png' border='0' alt="back"/></a><br/><br/>
<a href='https://postimg.org/image/eo34tv82v/' target='_blank'><img src='https://s7.postimg.org/dlsybbp9n/back_Night.png' border='0' alt="backNight"/></a><br/><br/>
<a href='https://postimg.org/image/s3q5jbgkn/' target='_blank'><img src='https://s7.postimg.org/wcuvlhju3/layer4.png' border='0' alt="layer4"/></a><br/><br/>

**Comment by tobyplowy** [comment]
@LordStephen77 having a game to test this would be useful but I will try my best with what I have 

(I have a question will random textures be supported this is important to get an answer to so I can make random textures beforehand for future use​)

Bdw did you get the leg animations from starbound?

**Comment by tobyplowy** [comment]
>(I have a question will random textures be supported this is important to get an answer to so I can make random textures beforehand for future use​)

Nevermind it seems you do support it

**Comment by LordStephen77** [comment]
@tobyplowy Textures I've found, I do not know, but I think they're strongly inspired by starbound and terrary. The blocks have reset them to 16x16 because 8x8 are too small and it is difficult to handle them. The final effect is better.

Textures I like with a 8-bit 80s style (starbound design). You decide how to do it, I've seen your MT texture pack and you're good at it.

Graphic priority for now are blocks (earth, grass, stone and more) because I have to test the map of play, then the rest comes later.

If you mean the procedural generation of the world, for now there will not be, but random maps will be taken from me, so to make the random effect. I then study the arguing that generates the maps (perlin) and implement it as I did for the day and night cycle.

Textures do not affect, they are only images that are chosen by the game engine based on a unique ID.

Thank you.

**Comment by tobyplowy** [comment]
> If you mean the procedural generation of the world, for now there will not be, but random maps will be taken from me, so to make the random effect. I then study the arguing that generates the maps (perlin) and implement it as I did for the day and night cycle.

No I mean to test the textures in game to see how they affect the world it's useful but not needed I can still make a mokup and test it that way

now my question about random textures is valid again so will there be randomized textures

Also will texturepack support be added in the future?
Also sorry about all the questions

**Comment by LordStephen77** [comment]
@tobyplowy I'm glad you ask questions, it's a good way to think about what to do and what not. Support for textures packs is not difficult. There will also be a small API available to create additional Java mods (simplified as a script language), this is the best choice because it will be sufficient to create the jar as if it were a small library simplifying all the work.

**Comment by tobyplowy** [comment]
> The blocks have reset them to 16x16

the textures you gave me have blocks that are 10x9 and 10x10

**Comment by tobyplowy** [comment]
> for textures packs is not difficult.

good to hear that

> There will also be a small API available to create additional Java mods

sounds good

**Comment by tobyplowy** [comment]
> > The blocks have reset them to 16x16 

the textures you gave me have blocks that are 10x9 and 10x10 not 16x16 and also not 8x8

I need an answer to the above Question to be able to do the art

**Comment by LordStephen77** [comment]
@tobyplowy I confirm. 16 x 16 pixels all blocks.

A good news! I managed to implement everything with javaFX, this is a good start, javaFX is very powerful, it is included in JDK 8 (no dependencies in the project) and are in collaboration with a student who develops a wrapper library that simplifies the realization of 2d games in JavaFX, simple and effective. Virtually just finished the game engine, I can start multiplayer sockets.

16x16 blocks like Minetest games. Blocks too small are difficult to handle in the game. I prefer 16 x 16 because I'm going to implement interesting features for building structures and other things for building and crafting (multi crafting).

If I reply late it means that I work on the engine of the game. I've made good progress, I'm anxious to finish the main game menu, and implement the first sounds and music.

Thanks :D

**Comment by tobyplowy** [comment]
@LordStephen77 so do I just redesign the block texture picture?

**Comment by tobyplowy** [comment]
@LordStephen77 also can every block be put in a new picture (it's better for work flow and for saving space)

**Comment by LordStephen77** [comment]
@tobyplowy You are right. Forgive me in a hurry I forgot the most obvious thing. You can create textures with separate images as per minetest. No sritesheet is needed, so it is much easier to handle the code, and it will be even easier to implement future textures packs.

**Comment by tobyplowy** [comment]
@LordStephen77 Ok I will get on it probably will be tomorrow but I may make something today

**Comment by tobyplowy** [comment]
@LordStephen77 can you add an new tile sistem
template (NOTE ALL TEXTURES USED IN THIS COMMENT ARE FOR DEMONSTATION PORPUSSUS THEY ARE BAD)

demo tile brakdouwn
![template](https://user-images.githubusercontent.com/16853304/27101983-eba6f10a-5083-11e7-98cf-d6ff6d38f77c.png)
what that shuld make
![tile pres](https://user-images.githubusercontent.com/16853304/27102012-0b66c290-5084-11e7-91f2-66e606ade1c8.png)
what can be made with it
![tile pres](https://user-images.githubusercontent.com/16853304/27102029-1fdc6fea-5084-11e7-87cd-534366464d84.png)
grass demo tiles
![template](https://user-images.githubusercontent.com/16853304/27102056-3307450e-5084-11e7-8f2d-7045baeb7066.png)

**Comment by tobyplowy** [comment]
@LordStephen77 I hope The Above comment issent too confuseing 

Some important notes

Only the complete red tile has a collision box the other ones are decorative

I copyed the same thing 5 times in the same image those are for random Textures (it shuld switch out the first one with one of the other 4)

**Comment by LordStephen77** [comment]
@tobyplowy Yes! The cartoon style I like. It's simple, and in the following days I'm sure you'll make some good material. The base above is great. I like the grass as you did it, Thick Terrarium Thickness. I want a different design from Minetest and Minecraft, you understand my idea well. Thanks so much. Work on these textures when you have time. I continue with the code.

I almost finished the main game menu (provisional, but it's important code, graphics are easy to implement).

**Comment by tobyplowy** [comment]
@LordStephen77 grass and dirt tiles
grass

![grass](https://user-images.githubusercontent.com/16853304/27130151-5e794792-5105-11e7-9b0d-d1c1a4551d62.png)
dirt
![dirt](https://user-images.githubusercontent.com/16853304/27130150-5e5eea00-5105-11e7-9759-8e589a009dbe.png)
this is what they look like if used correctly
![lolol](https://user-images.githubusercontent.com/16853304/27130226-bcaf23a4-5105-11e7-8f5c-812d4fd6c3b4.png)

NOTE the random dirt and grass :D

**Comment by tobyplowy** [comment]
@LordStephen77 I will make more but can you give me some feedback also Can you possibly show this in game?

**Comment by LordStephen77** [comment]
@tobyplowy All right. Expect to release a first demo version. There are bugs I have to solve but everything works, apart from the torches. The structure is very similar to Minetest, Minecraft, and I'm adding elements of Terraria and other historical games.

The game is alive, but it does work a lot to improve the engine. Next, the next step is adding features.

**Comment by tobyplowy** [comment]
@LordStephen77 are the textures good?

**Comment by LordStephen77** [comment]
@tobyplowy Yes, I've converted them with my python script, now all 8 bit super pixeled. I have succeeded in completing the following things:

- Activate the F key to display real-time fps in real time with busy and free memory.
- Generation of casual infinite world completed and perfectly optimized. After several searches I solved.
- Optimization 8x8 textures and hybrid Minetcraft and Terror hybrid design, lighting effects, improved daytime night cycle (on 4 progressive cycles).
- Auto save saves a .sav file in the game folder, and loads button to load the last world.
- Crafting and inventory system implemented, I just need to fix small bugs but it works.
- All the code is organized and commented in an orderly way, I have already prepared the way for the textures pack.
- Other things I'm fixing.

That's right. I do everything possible to finish the last thing, publicize the executable jar and try the game so you have the clearest ideas on how I want to make it. Thanks so much.

**Comment by tobyplowy** [comment]
@LordStephen77 well it's crazy how much work you've already done but watch out don't get burnt out because if you spend too much time on this projects in a small amount of time you will get tired of it very fast so take couple breaks every now and then ok ;)

Also can you make trees work like this 
https://m.youtube.com/watch?v=sRqPiD2XKOk
In this video from Starbound you see how the tree falls over and it collapses into wood 
(I think this looks fantastic)
NOTE the tree is not made of blocks it's an object (it's made of a copple objects)

**Comment by LordStephen77** [comment]
@tobyplowy Quiet I'm crazy!

Now if you sow the ground sapling, they create the trees. Trees emit sun rays (old style effect as you like me). I fixed some things. Then I write the changelog at each release.

Trees can be used to climb (for now it works as underwater). Water added to the generation of random worlds. Gradual light effect when digging the blocks. Minerals added and dop blocks and minerals added. I've activated everything I did. There are only a few corrections.

I just send the commit with all the code. I also attach the executable jar. Soon, I end up doing so all the work.

**Comment by tobyplowy** [comment]
@LordStephen77 it was really hard to read that :/
But I think I understand what you said

**Comment by tobyplowy** [comment]
@LordStephen77 can you tell me a date when the .jar file will be available?

**Comment by tobyplowy** [comment]
cobblestone
![cobblestone](https://user-images.githubusercontent.com/16853304/27150053-03a2f61a-5146-11e7-9e21-527d638bbf3f.png)
stone (this is a wip texture)
![stone](https://user-images.githubusercontent.com/16853304/27150060-095c05ec-5146-11e7-8ec9-b1ee16ea0420.png)

**Comment by LordStephen77** [comment]
@tobyplowy I have published all the project in the repository, the jar will be within 1 hour maximum 2, I will also include game instructions, screenshots and instructions to build the terminal build (windows, linux, mac os). The textures you've made I have to work for now look at what I've edited. They are perfect as a style. Yours is suitable for new content, biomes, and additional things. Hope you like the game even if it's still an unstable alpha.

**Comment by tobyplowy** [comment]
> Hope you like the game even if it's still an unstable alpha.

I can Look past bugs even Better is that i really like unstable games it shows The development of the complete project also I like to bug test and brake games :)

I have seen some weird shit in the 18 years I have been gaming so dount worry :)

**Comment by tobyplowy** [comment]
@LordStephen77 i have 2 questions

will texturepack support be in the first build?

Will you be able to make background blocks

**Comment by LordStephen77** [comment]
I will also make the multiplayer. The ability to apply textures packs, and see what to implement later.

Please read below the changelog. I kindly ask you to test all the new keyboard features (shortcuts), check out any graphic details and bugs, so open a bugs topic here (clearly specifying what you like and what not). Thanks so much.

Changelog 14/06/2017 (european date format)

- Optimized and rewritten game engine.
- Added lighting effects to trees, and blocks.
- Progressive day and night cycle on 5 levels.
- Sapling work. Now the trees grow.
- Added oxygen bar and the ability to dive into the sea.
- Map generator, small, medium, and large. Maps are generated optimally, including minerals based on minimum depth detection parameters.
- Added keyboard features: play pause (P), music on / off (M), zoom increase (+), zoom reduction (-), zoom default (O), framerate display and free memory (F), Climp (W), inventory system + Crafting (E).
- Mouse wheel to select hotbar boxes. Select hotbar boxes with 1-0 keys.
- Mineral drops and improved blocks.
- Some bug fixes, and other code improvements.

Pre-Alpha - Executable (JAR Download):

https://drive.google.com/open?id=0B4XH4Bj8pv4sSTNxWkw5UkRvQVU

**Comment by LordStephen77** [comment]
@tobyplowy I forgot. Now that you have the first pre-release executable, you have a clearer idea of my project and what I want to do (minecraft + terraria and starbound) now the game engine resembles a 2d minetest, with typical terrarios and starbound . I have to work to fix some bugs. As soon as I fully implement the code I wrote for multi-crafting (no sandbox game has this thing, it's my idea) if you've noticed, moving the cursor you have 2 box-selection (blue and red). If you have a tree or blocks in front of your box-selection do not cross the blocks but follow their structure. That red places blocks, objects and the classic crafting table, the blue one serves for the second crafting scheme completely different from the other.

See you soon. Just find time working to improve this project more and more. Then I'll think of the small official website to promote it. Thanks so much. Your interest moves me to do better and better.

**Comment by tobyplowy** [comment]
@LordStephen77 so im still testing but i alredy fund a small problem, 
the leafs are in my way and i have to barake them to get past

**Comment by tobyplowy** [comment]
@LordStephen77 maybe also make the player be able to jump 2 blocks instead of 1 wuld be good

**Comment by tobyplowy** [comment]
@LordStephen77 at night its compleatly black i see noting anymore maybe make it so you can still see alittle bit

**Comment by tobyplowy** [comment]
zoom reduction (-) works but for zoom increase (+) i have to hold shift so that is (hold shift and press + for zoom in)

**Comment by tobyplowy** [comment]
pause (P) doesent work also i think you shuld put that on escape (esc by standart)

**Comment by tobyplowy** [comment]
(O) also doesent seem to do anyting

**Comment by azekillDIABLO** [comment]
wow this is nice! i would ike to help! :smirk:

**Comment by LordStephen77** [comment]
@tobyplowy With + increase the zoom, decrease it without shift. The or is used to return to the default zoom. I have to fix it. And many other things.

For bugs you find you make a unique message so copy and paste into the clipboard and check the code.

I hope you like the job. Even if it presents bugs I really like the end result.

For crafting is still experimental, in just 7 days I did too much. Thank you.

I can release the game in .exe for windows. It can be useful.

AzekillDIABLO I'm glad you're here. I need a Java socket programmer. Or a standard java programmer that helps me optimize code and fix bugs and more. The project is all in the repository so you can work it if you want, then I'll tell you how to share the code with me to transfer the commit.

Thank you.

**Comment by azekillDIABLO** [comment]
ok, i call you back in two month, i go learn Java. :+1: 

no i was still about your website... i really wanna do it!

**Comment by LordStephen77** [comment]
@azekillDIABLO If you want to learn how to make 2d games I recommend javascript, use Phaser.js is a very easy to use, super-documented and powerful game engine to play games for all gadgets (windows, Linux, Mac OS, Android, iOS and Web). Let go of Java, it is a very powerful language and has an immense standard library. But it is also very abnormal at times.

**Comment by tobyplowy** [comment]
@LordStephen77 it maybe a good idea to tell about this project on the terraria and Starbound and maybe the Minecraft forums to get more poeple that may be interested in helping (it will make development go super fast)

**Comment by LordStephen77** [comment]
@tobyplowy Yes you are right. I dedicate myself to my old projects, I want to publish them here on github and terminate them. I do my best to find code collaborators to simplify the development of this game. If you know a Java programmer you know where I am. The code is pretty easy so you do not need an expert programmer to work on it. Thank you.

**Comment by tobyplowy** [comment]
@LordStephen77 when will the texturepack support and the modding engine be added (I think you have to do this early in development so you don't have to do it later because later it will be really problematic to add it)

**Comment by LordStephen77** [comment]
@tobyplowy No, that's the reverse. First I have to fix all bugs, improve code, and fix everything. Then I then implement new features. A small modding API is already doing, as soon as the audience ends. Thank you.

**Comment by Gerold55** [comment]
Can I please help with inventory's and with some other textures?

**Comment by LordStephen77** [comment]
@Gerold55 Thank you so much for your interest. For now I work on code enhancement, bug fixing, and optimization. As soon as I set up the project I will see how to define the graphics and all the necessary elements. Thanks so much.
